catalog/app/containers/Bucket/Queries/Queries.tsx

```diff
@@ -0,0 +1,249 @@
+import * as R from 'ramda'
+import * as React from 'react'
+import { RouteComponentProps } from 'react-router'
+import * as M from '@material-ui/core'
+import * as Lab from '@material-ui/lab'
+
+import { docs } from 'constants/urls'
+import StyledLink from 'utils/StyledLink'
+
+import QueryResult from './QueryResult'
+import QuerySelect from './QuerySelect'
+import QueryViewer from './QueryViewer'
+import * as requests from './requests'
+
+const useStyles = M.makeStyles((t) => ({
+  actions: {
+    margin: t.spacing(2, 0),
+  },
+  container: {
+    padding: t.spacing(3),
+  },
+  inner: {
+    margin: t.spacing(2, 0, 0),
+  },
+  form: {
+    margin: t.spacing(0, 0, 4),
+  },
+  select: {
+    margin: t.spacing(3, 0),
+  },
+  viewer: {
+    margin: t.spacing(3, 0),
+  },
+}))
+
+interface SearchResultsFetcherProps {
+  children: (
+    props: requests.AsyncData<requests.ElasticSearchResults>,
+  ) => React.ReactElement
+  queryBody: requests.ElasticSearchQuery | string
+}
+
+function SearchResultsFetcher({ children, queryBody }: SearchResultsFetcherProps) {
+  const resultsData = requests.useSearch(queryBody)
+  return children(resultsData)
+}
+
+interface QueryFetcherProps {
+  children: (props: requests.AsyncData<requests.ElasticSearchQuery>) => React.ReactElement
+  query: requests.Query | null
+}
+
+function QueryFetcher({ children, query }: QueryFetcherProps) {
+  const queryUrl = React.useMemo(() => (query ? query.url : ''), [query])
+  const queryData = requests.useQuery(queryUrl)
+  return children(queryData)
+}
+
+interface QueriesStatePropsInjectProps {
+  customQueryBody: requests.ElasticSearchQuery | string
+  error: Error | null
+  handleError: (error: Error | null) => void
+  handleQueryBodyChange: (q: requests.ElasticSearchQuery | null) => void
+  handleQueryMetaChange: (q: requests.Query | null) => void
+  handleSubmit: (q: requests.ElasticSearchQuery | string) => () => void
+  queries: requests.Query[]
+  queryData: requests.AsyncData<requests.ElasticSearchQuery>
+  queryMeta: requests.Query | null
+  resultsData: requests.AsyncData<requests.ElasticSearchResults>
+}
+
+interface QueriesStateProps {
+  bucket: string
+  children: (props: QueriesStatePropsInjectProps) => React.ReactElement
+}
+
+function NoQueries() {
+  return (
+    <M.Box pt={5} textAlign="center">
+      <M.Typography variant="h4">No queries</M.Typography>
+      <M.Box pt={2} />
+      <M.Typography>
+        Add queries to config according to{' '}
```

Comment:
shouldnt we allow running arbitrary queries even when there's no config? like, just show the text box

Reply:
Yes, we should. I'll make changes